### PR TITLE
EL-4087 - Marquee - Wizard Custom Step Content

### PR DESCRIPTION
**Pull request by ashley-hunter**
#### Checklist

* [x] The contents of this PR are mine to submit. No third party code or other material is being committed.
* [x] New third party dependencies (if any) are linked via `package.json`. Third party dependencies are licensed under one of the following: Apache, MIT, BSD, Mozilla Public License, Eclipse Public License, or Oracle Binary Code License.
* [x] The code in this PR does not contain export-restricted encryption algorithms.

#### Ticket / Issue

https://portal.digitalsafe.net/browse/EL-4087

#### Description of Proposed Changes

- Adding a `context` input on each step to allow them to provide custom data to the step template
- Adding `index` and `context` properties to the step template context
- Improving documentation with new examples and documenting the new context interface and input
- Added test case
- Fixed an incorrect type used by a splitter event

#### Documentation CI URL

https://pages.github.houston.softwaregrp.net/sepg-docs-qa/UXAspects_CI_UXAspects_EL-4087-Marquee-Wizard-Custom-Content

#### Downstream Build(s)

* [caf/ux-aspects-micro-focus](https://jenkins.swinfra.net/job/SEPG/job/caf/job/caf~ux-aspects-micro-focus~EL-4087-Marquee-Wizard-Custom-Content~CI/)

### CODE REVIEW

**Review by donna-marijne — CHANGES_REQUESTED** [review]
Thanks, this looks good. Apart from the badge issue, just a couple of documentation suggestions.

**Comment by ashley-hunter** [comment]
I have made the suggested review fixes and also updated the badge directive to accept numbers as inputs rather than solely strings. I have added test cases for this change also.

File: docs/app/pages/components/components-sections/wizard/marquee-wizard/snippets/step-template.html [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<ux-marquee-wizard [stepTemplate]="stepTemplate">
+    <ng-template #stepTemplate
+                 let-step
+                 let-index="index"
+                 let-context="context">
+
+                    <span class="marquee-wizard-step-title"
```
Comment (donna-marijne) [inline docs/app/pages/components/components-sections/wizard/marquee-wizard/snippets/step-template.html:7]: Indentation issue here.

**Review by donna-marijne — CHANGES_REQUESTED** [review]
Thank you, great to get the badge issue taken care of here, it will be a much better experience for anyone trying to use it. There's just one indentation issue in the code snippet.

**Comment by ashley-hunter** [comment]
Thanks Donna, indentation should now be corrected

**Review by donna-marijne — DISMISSED** [review]
Pushed a fix. @Sdhilipkumar waiting on your review now.

**Review by donna-marijne — APPROVED** [review]

**Review by Sdhilipkumar — APPROVED** [review]
Looks good.

**Pull request merged**